Lib/asyncio/sslproto.py

```diff
@@ -555,6 +558,12 @@ def _start_handshake(self):
         # the SSL handshake
         self._write_backlog.append((b'', 1))
         self._loop.call_soon(self._process_write_backlog)
+        self._loop.call_later(self._ssl_handshake_timeout, self._check_handshake_timeout)
```

Comment:
You should save a reference to this handle (`self._handshake_timeout_handle = self._loop.call_later(...)`) and cancel it (`self. _handshake_timeout_handle.cancel()`) if the handshake is successful.